src/reconcile/rec_row.c

```diff
@@ -53,23 +53,125 @@ __rec_key_state_update(WTI_RECONCILE *r, bool ovfl_key)
 
 /*
  * __rec_cell_build_int_key --
- *     Process a key and return a WT_CELL structure and byte string to be stored on a row-store
- *     internal page.
+ *     Build an internal key cell and update reconcile buffers as needed.
+ *
+ * If 'r' is non-NULL, both r->cur and r->k.buf are updated. If 'r' is NULL, only the provided 'key'
+ *     buffer is updated.
  */
 static int
-__rec_cell_build_int_key(WT_SESSION_IMPL *session, WTI_RECONCILE *r, const void *data, size_t size)
+__rec_cell_build_int_key(WT_SESSION_IMPL *session,
+  WTI_RECONCILE *r, /* Optional reconcile structure */
```

Comment:
Why we still have this? Have we discussed to get rid of this?

Reply:
I just don't like to have WT_RECONCILE here. The logic is simple but it is a breach of encapsulation. We should not depend on anything that is in the reconcile module.